framework/include/utils/MultiMooseEnum.h

```diff
@@ -27,29 +27,48 @@ typedef std::vector<MooseEnumItem>::const_iterator MooseEnumIterator;
 
 /**
  * This is a "smart" enum class intended to replace many of the
- * shortcomings in the C++ enum type It should be initialized with a
- * comma-separated list of strings which become the enum values.  You
- * may also optionally supply numeric ints for one or more values
- * similar to a C++ enum.  This is done with the "=" sign. It can be
- * used any place where an integer (switch statements), const char* or
- * std::string is expected.  In addition the InputParameters system
- * has full support for this Enum type
+ * shortcomings in the C++ enum type. There are two parts to a MultiMooseEnum:
+ * 1) The list of all possible values that a variable can have and
+ * 2) The subset of these values that the variable is actually set to.
+ * It should be initialized with a comma-separated list of strings which
+ * become the possible enum values. You may also optionally supply numeric
+ * ints for one or more values similar to a C++ enum. This is done with the
+ * "=" sign. It can be used any place where an integer (switch statements),
+ * const char* or * std::string is expected. In addition the
+ * InputParameters system has full support for this Enum type.
  */
 class MultiMooseEnum : public MooseEnumBase
 {
 public:
+  ///@{
   /**
-   * Constructor that takes a list of enumeration values, and a separate string to set a default for
-   * this instance
-   * @param names - a list of names for this enumeration
-   * @param default_names - the default value for this enumeration instance
+   * Constructor that takes a list of valid enumeration values, and a separate string to set default
+   * values for this instance
+   * @param valid_names - a list of all possible values (names) for this enumeration
+   * @param initialization_values - the value(s) to set this enumeration instance to
    * @param allow_out_of_range - determines whether this enumeration will accept values outside of
-   * it's range of
-   *                       defined values.
+   * it's range of initially defined values, allowing for the possiblity to add additional valid
+   * values to an object after it has been initialized.
```

Comment:
maybe split the allowing into a different sentence for clarity?
```suggestion
   * its range of initially defined values, allowing for the possiblity to add additional valid
   * values to an object after it has been initialized.
```